src/types.php

```diff
@@ -308,6 +308,28 @@ public static function iterable(Type $keyType = self::mixed, Type $valueType = s
         return new IterableType($keyType, $valueType);
     }
 
+    /**
+     * @psalm-pure
+     * @param array<string, Type|Property> $properties
+     */
+    public static function objectShape(array $properties = []): ObjectShapeType
+    {
+        return new ObjectShapeType(
+            array_map(
+                static fn (Type|Property $property): Property => $property instanceof Type ? new Property($property) : $property,
+                $properties,
+            ),
+        );
+    }
+
+    /**
+     * @psalm-pure
+     */
+    public static function prop(Type $type, bool $optional = false): Property
```

Comment:
Should we use full names here?..

Reply:
I can't say why, but I like the full name.
Maybe `prop` sounds like `a` or `b` for variable name.

Reply:
Okay, agree. I will merge it here and then change all short names in master.